utils/queue/priorityQueue.go

```diff
@@ -0,0 +1,84 @@
+package queue
+
+import (
+	"container/heap"
+	"time"
+)
+
+// Item stores the attributes which will be pushed to the priority queue..
+type Item[T any] struct {
+	Value     T
+	Priority  int
+	timeStamp int64
+	index     int
+}
+
+// PriorityQueue ....
```

Comment:
So meaningful :laughing:

Reply:
Given that as you put it it's very close to the heap std implementation, I think it would be worth explaining in a comment how it is different from a `std` one and when to use this implementation.

Reply:
There is no std implementation of a priority queue, at least none that I am aware of. There is only an [example](https://pkg.go.dev/container/heap#example-package-PriorityQueue) in the docs. The only standard is the `heap.Interface` that types must implement to be used for heap operations.